WordPress/gutenberg
# Add section styles switch button in block toolbar in zoom out mode

## What?
Add section styles switch button in block toolbar in zoom-out mode.

## Why?
Fixes: https://github.com/WordPress/gutenberg/issues/66465

## How?
Adding the button to the toolbar.

## Testing Instructions
1. Browser to the editor and open the inserter.
2. Add a pattern.
3. If you are not in the zoom-out view select the zoom-out view.
4. Select the just-added pattern and observe the toolbar. If the block has block style variations available, it should render a new button with the 'Swtich Style' label.
5. If the block has block variations available, each click on the 'Switch style' button should change the style to the next variation.

## Screenshots or screencast 

https://github.com/user-attachments/assets/ede56aec-a6db-48f1-9f73-aea007004e31
